src/logger.cpp

```diff
@@ -10,21 +10,19 @@
 namespace diskann
 {
 
+#ifdef ENABLE_CUSTOM_LOGGER
 DISKANN_DLLEXPORT ANNStreamBuf coutBuff(stdout);
 DISKANN_DLLEXPORT ANNStreamBuf cerrBuff(stderr);
 
 DISKANN_DLLEXPORT std::basic_ostream<char> cout(&coutBuff);
 DISKANN_DLLEXPORT std::basic_ostream<char> cerr(&cerrBuff);
```

Comment:
The intention of ENABLE_CUSTOM_LOGGER is to allow prod env set a custom logger in `g_logger.`

The default behavior (i.e., without ENABLE_CUSTOM_LOGGER) is to use stdout and stderror logging which is what the line 14-line18 does?

So those global variables seems to be an issue?

Reply:
Correct, the global/statically initialized diskann::{cout,cerr} impls we've created are the progenitor of issues that we *see* in python but probably also exist everywhere. 

As I mentioned in a response to @yiylin in this PR, I had assumed we could just safely rely on std::{cout,cerr} instead, but it would mean that the custom logger would most likely continue having the problem. I was very hesitant to update any code that existed in prod, though - especially without knowing why it was the way it was.